server/graphql-schema.js

```diff
@@ -197,46 +222,94 @@ const graphqlSchema = gql`
     }
 
     input TestPlanReportInput {
-        testPlan: ID!
-        testPlanTarget: ID!
+        testPlanVersionId: ID!
+        testPlanTarget: TestPlanTargetInput!
+    }
+
+    input LocationOfDataInput {
+        testPlanId: ID
+        testPlanVersionId: ID
+        testIndex: Int
+        passThroughIndex: Int
+        testPlanReportId: ID
+        testPlanTargetId: ID
+        browserId: ID
+        browserVersion: String
+        atId: ID
+        atVersion: String
+        testPlanRunId: ID
+        testResultIndex: Int
+        passThroughResultIndex: Int
+    }
+    type LocationOfData {
+        testPlanId: ID
+        testPlanVersionId: ID
+        testIndex: Int
+        passThroughIndex: Int
+        testPlanReportId: ID
+        testPlanTargetId: ID
+        browserId: ID
+        browserVersion: String
+        atId: ID
+        atVersion: String
+        testPlanRunId: ID
+        testResultIndex: Int
+        passThroughResultIndex: Int
+    }
+
+    """
+    All the records that are associated with the id(s) of a given locationOfData
+    """
+    type PopulatedData {
+        locationOfData: LocationOfData!
```

Comment:
@alflennik can you explain LocationOfData? What does that mean?

Reply:
Sure - the PR description has a basic description, but let me describe in more in depth.

It's actually quite common through the app to need to point to a particular piece of data. We often need to say something like, "I want this TestPlanRun on this TestPlanReport on this TestPlanVersion on this TestPlan".

An example of a pressing use case for this PR is the testPlanTarget, atVersion and browserVersion which are associated with a TestPlanReport. When we create a new TestPlanReport for the Test Queue, I wanted to provide a list of data which was newly created. That list might look like this:

```js
[
  { testPlanReportId: 12 },
  { testPlanReportId: 12, testPlanTargetId: 3 },
  { testPlanReportId: 12, testPlanTargetId: 3, atVersion: '2021.2' }
]
```

This means that we created 3 new records, a TestPlanReport, TestPlanTarget and AtVersion, and using this list we could communicate that information to the user.

And GraphQL can automatically take one of those small little objects like `{ testPlanReportId: 12, testPlanTargetId: 3, atVersion: '2021.2' }` and just by traversing the associations it can tell you all the data which is implied by those IDs: the TestPlan, the TestPlanVersion, the TestPlanReport, the TestPlanTarget, and the AtVersion.

Looking a bit into the future, I can see this being useful for stuff like the Test Reports pages, where we need to support navigating a giant tree of reports, especially when we need to load a bunch of data just from a given URL. And the idea originally came to me when I realized I needed it for expressing conflicts between TestPlanRuns.

Hopefully that helps!